File: godfather/roles/town/super_saint.py
import asyncio
from godfather.roles.base import Role
from godfather.roles.mixins import Townie

DESCRIPTION = 'You will blow up the last person to lynch you!'


class SuperSaint(Townie, Role):
    name = 'Super Saint'
    description = DESCRIPTION

    async def on_lynch(self, game, player):
        last_voted = game.votes[player.user.id][-1]
        last_voted.is_alive = False
        async with game.channel.typing():
            await game.channel.send('💣 **BOOOOOOOOOOOOOOM!!!**')
            await asyncio.sleep(2)
            await game.channel.send(f'{last_voted.user} hammered the super saint and was blown up! He was a *{last_voted.display_role}*')

File: src/lib/mafia/roles/town/Super_Saint.ts
import Townie from '@mafia/mixins/Townie';
import Role from '@mafia/structures/Role';
import { sleep } from '@util/utils';
import type Player from '@mafia/structures/Player';

class SuperSaint extends Role {
	public name = 'Super Saint';

	public constructor(player: Player) {
		super(player);
		this.description = this.game.t('roles/town:superSaintDescription');
	}

	public async onDeath() {
		if (this.player.deathReason.startsWith('eliminated')) {
			const { game } = this.player;

			const votesOnSaint = game.votes.on(this.player);
			const { by: lastVoter } = votesOnSaint!.slice().pop()!;

			await lastVoter.kill(this.game.t('roles/town:superSaintDeathReason'));
			await game.channel.send(this.game.t('roles/town:superSaintBoom'));
			await sleep(2 * 1000);

			await game.channel.send(`${this.game.t('roles/town:superSaintAnnouncement', { player: lastVoter })} ${lastVoter.displayRoleAndWill()}`);
		}

		return super.onDeath();
	}
}

export default Townie(SuperSaint);
